src/terminal/adapter/DispatchTypes.hpp

```diff
@@ -179,18 +179,28 @@ namespace Microsoft::Console::VirtualTerminal
         template<typename T>
         bool for_each(const T&& predicate) const
         {
+            auto values = _values;
+
             // We always return at least 1 value here, since an empty parameter
             // list is the equivalent of a single "default" parameter.
-            auto success = predicate(at(0));
-            for (auto i = 1u; i < _values.size(); i++)
+            if (values.empty())
             {
-                success = predicate(_values[i]) && success;
+                values = defaultParameters;
+            }
+
+            auto success = true;
+            for (const auto& v : values)
+            {
+                success = predicate(v) && success;
             }
             return success;
         }
 
     private:
-        gsl::span<const VTParameter> _values;
+        static constexpr VTParameter defaultParameter;
+        static constexpr std::span defaultParameters{ &defaultParameter, 1 };
+
+        std::span<const VTParameter> _values;

```

Comment:
This is a more significant change, as the `[]` operator of `std::span` is not considered safe anymore (unlike `gsl::span` it doesn't fail-fast). When fixing it with `til::at` I noticed that `for_each` can be simplified with a for-range loop.